src/Type.h

```diff
@@ -142,20 +142,46 @@ template<> struct halide_c_type_to_name<float> { static const bool known_type =
 template<> struct halide_c_type_to_name<double> { static const bool known_type = true; static halide_cplusplus_type_name name() { return { halide_cplusplus_type_name::Simple,  "double"}; } };
 template<> struct halide_c_type_to_name<struct buffer_t> { static const bool known_type = true; static halide_cplusplus_type_name name() { return { halide_cplusplus_type_name::Struct,  "buffer_t"}; } };
 
+// Default case (should be only Unknown types, since we specialize for Known types below).
```

Comment:
Per conversation, please add a few lines on how users can specialize the template to make their types known or a TODO for me to do so later.